### ISSUE TITLE
Addition of Sequence Number in QuoteTick

### ISSUE THREAD

**Issue by rsmb7z** [nautechsystems/nautilus_trader#1632]
# Feature Request

Currently, while we have the `trade_id` field to identify unique trades in TradeTick, there's no unique identifier for QuoteTick. This absence becomes critical when multiple ticks occur simultaneously, necessitating the maintenance of data order.

To address this, I propose adding a sequence number field (preferably integer) to the QuoteTick object. This addition would significantly enhance our ability to maintain data order, particularly in scenarios where multiple ticks occur simultaneously.

**Comment by cjdsellers** [comment]
Adding a field to `QuoteTick` would be a large breaking change, and a sequence number for top-of-book data may not fit a use case for the majority of users.

Could you provide a concrete example of where multiple quotes could occur for a single instrument at the same nanosecond, with differing prices or sizes?

**Comment by rsmb7z** [comment]
The idea sparked while setting primary keys in TimescaleDB. I discovered duplicate QuoteTicks, yet they shared identical prices, and size possibly owing to the structure of the TBBO schema.

Acknowledging the unlikelihood and negligible advantage, I'm closing the issue.

However, in exploring this, I stumbled upon another intriguing observation: sequence numbers may duplicate across multiple trades with varying prices and sizes. Thus, they can't serve as a direct unique identifier in TradeTick.
![image](https://github.com/nautechsystems/nautilus_trader/assets/105105941/9debfa8f-cbca-4770-a2eb-3c3a186bd380)
